### DISCUSSION TITLE
Chat Persistence Client use cases

### DISCUSSION THREAD

**Discussion by Abacaxi-Nelson** [GetStream/stream-chat-flutter#379] [General]
Hi all,

Trying to understand which use case "Chat Persistence Client" is covering:
- Offline chat ?
- Better performance ?

We are going to have large group conversation, and im thinking in adding local persistence !

Please share feedback :)
Nelson

**Comment by imtoori** [comment]
Hey Nelson,

The persistence client is useful for both the use cases!
It doesn't give you a full offline experience, but it covers most of the offline chat functionalities you need:
- channel and message query
- sending/updating/deleting messages 
- retrying those messages once you get the connection back

For sure, it gives you better performances because you'll always query the offline storage before performing the API call

for example:
- the `ChannelListView` will load immediately using the channels you have offline while calling the API and updating it accordingly
- the `MessageListView` will use the messages you have in the offline storage and will not query the channel if you have enough messages for the next page. Once you reach the oldest message you have in the offline storage for that channel, it will call the API to check if more messages are available

ps: hope you don't mind I converted the issue to a discussion, we're trying this new section for this kind of questions 🙂

**Comment by Abacaxi-Nelson** [comment]
hi @imtoori 
thanks for the explanation, helps a lot !
last question: what is the difference between regular and background mode ?

Thanks,
have a nice day

**Reply by xsahil03x** [reply]
Hey @Abacaxi-Nelson,

If you check the enum ConnectionMode it states that 

```dart
/// Various connection modes on which [StreamChatPersistenceClient] can work
enum ConnectionMode {
  /// Connects the [StreamChatPersistenceClient] on a regular/default isolate
  regular,

  /// Connects the [StreamChatPersistenceClient] on a background isolate
  background,
}
```
if you connect your persistence client using the background mode it will run all its queries on a separate 
isolate leaving the main isolate free for other tasks.

Thanks

**Comment by Abacaxi-Nelson** [comment]
thanks @imtoori @xsahil03x
